### PR TITLE
Fix product's feature selection

### LINKED ISSUE #16306: Improve feature creation in product page
#### Describe the bug

> Edited by @HanaRebaiQA 

There are two cases in product page : 

**Product V1 page** : There is no possibility to delete a feature when we put `Choose a value` and `Choose a feature`. The features are duplicated after save and when we put a customized value for it.

Here is a screen-record : 8.0.x with product V1 page

https://user-images.githubusercontent.com/92912932/177517930-35644b44-8d22-43f1-8c9e-83ed0550d8cc.mp4

**Product V2 page** : An exception is thrown when we put `choose a value` and `choose a feature ` 

Here is a screen-record : 8.0.x with product V2 page

https://user-images.githubusercontent.com/92912932/177517938-e58de612-24d2-4aeb-8da9-f3ba98340fee.mp4

>  //*************************The original Description************************//

If you create a feature in a product page "Add a new feature" and give it a custom value, **but without selecting an existing feature** then it is impossible to delete it after

In addition to no being able to delete it, it seems to impact existing and fine features by applying the custom value of the created feature to all the features' values.

#### Expected behavior
When you add a feature, you have a drop-down with all features. 

If you have not yet selected a feature, the “pre-defined value” drop-down is disabled and the “customized value” field should be also disabled (currently it’s not).

capture d'écran_2129

If you select a pre-defined value, the customized value should be disabled because a feature can’t have both.

capture d'écran_2130

If you start enter a customized value, the pre-defined value drop-down should be disabled because a feature can’t have both.

capture d'écran_2131

#### Steps to Reproduce

1. Go to 'Catalog > products > select any product'
2. In the product page, on the first tab, click on 'Add a feature'
3. For "Feature" and "Pre-defined value", select "Choose a feature" and add a custom value
4. Save
5. go to another page and come back to the product page
6. See the feature you created, delete it and save
7. Go to another page and come back, the feature still exist.

**Screenshots**

See the video:
https://drive.google.com/file/d/1nZPWLhODwa9FQlvmK2aHRDWNPVEy8O1W/view

#### Additional informations
PrestaShop version: 1.7.7.0
PHP version: 7.2

### PR DESCRIPTION
**Pull request by alfonsolr09**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      |  We make the boxes are deactivated if we have not selected the characteristic. If we add a custom value, the value box is disabled. If we select a value in the list, the custom value box is deactivated and deletes the value (if there was something written)
| Type?             | bug fix 
| Category?         |  BO 
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Fixes #16306
| How to test?      | See features in product page
| Possible impacts? | No

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27097)

### CODE REVIEW

**Comment by alfonsolr09** [comment]
@atomiix now??

**Comment by NeOMakinG** [comment]
@alfonsolr09 is it related to the product page v2?

**Comment by alfonsolr09** [comment]
Nop, v1

**Comment by alfonsolr09** [comment]
what happens here? 😊

**Review by NeOMakinG — APPROVED** [review]
I've nothing against it honestly, but this page should be removed in next version

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 1]
```diff
@@ -426,17 +430,43 @@ const featuresCollection = (function () {
...
+        const that = event.currentTarget;
+        const $row = $($(that).parents('.row')[0]);
+
+        const $customValueSelector = $row.find('input[type=text]');
+        $customValueSelector.val('');
+      });
+
+      $(document).on('keyup', '.feature-collection input[type=text]', (event) => {
+        const that = event.currentTarget;
+        const $row = $($(that).parents('.row')[0]);
```
Comment (matks) [inline admin-dev/themes/default/js/bundle/product/form.js:444]: @NeOMakinG should we handle the case where there is no offset zero? should we not verify the DOM structure?

Comment (NeOMakinG) [reply admin-dev/themes/default/js/bundle/product/form.js:444]: Is there a case where the dom structure would be different? I don't think so, this is why I didn't comment on this

But if you want to verify, feel free, shouldn't add too much pain and problems

**Review by sowbiba — APPROVED** [review]

**Comment by sowbiba** [comment]
Ping @jolelievre 
Is this fix useful on develop ?

**Review by sowbiba — APPROVED** [review]

**Comment by jolelievre** [comment]
Hi @sowbiba and @alfonsolr09 

this fix targets the legacy page which is gonna be removed in 8.0 so there's no real interest in fixing this bug here since it will never be released. Though you can check if the issue mentioned is still a problem on the experimental page. If so it should be fixed in the new page, if not the issue can be closed and so can this PR.

**Comment by alfonsolr09** [comment]
> Hi @sowbiba and @alfonsolr09 
> 
> this fix targets the legacy page which is gonna be removed in 8.0 so there's no real interest in fixing this bug here since it will never be released. Though you can check if the issue mentioned is still a problem on the experimental page. If so it should be fixed in the new page, if not the issue can be closed and so can this PR.

Why didn't you let me know before solving the problem?
Can you test if the error is fixed in the new version?

**Comment by marwachelly** [comment]
Hello,

I tested the PR and I noticed  the following points:

- The first required behavior is ok 

**First expected part**
image

**with PR**
image

- the expected behavior for the second and the third part described in the issue is not implemented accordingly

**Second expected part**
image

I observed the following incorrect behavior:

**With PR**
image
image

Could you please check?

Many thanks,

**Review by marwachelly — CHANGES_REQUESTED** [review]
Hello,

I tested the PR and I noticed the following points:

The first required behavior is ok
**First expected part**
image

**with PR**
image

the expected behavior for the second and the third part described in the issue is not implemented accordingly
**Second expected part**
image

I observed the following incorrect behavior:

**With PR**
image

image


Could you please check?

Many thanks,

**Comment by PierreRambaud** [comment]
Hi @alfonsolr09 ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Comment by alfonsolr09** [comment]
> Hola,
> 
> Probé el PR y noté los siguientes puntos:
> 
> El primer comportamiento requerido está bien **Primera parte esperada** imagen
> 
> **con relaciones públicas** imagen
> 
> el comportamiento esperado para la segunda y la tercera parte descrita en el problema no se implementa en consecuencia **Segunda parte esperada** imagen
> 
> Observé el siguiente comportamiento incorrecto:
> 
> **con relaciones públicas** imagen
> 
> imagen
> 
> ¿Podrías comprobarlo?
> 
> Muchas gracias,

i 

> Hello,
> 
> I tested the PR and I noticed the following points:
> 
> The first required behavior is ok **First expected part** image
> 
> **with PR** image
> 
> the expected behavior for the second and the third part described in the issue is not implemented accordingly **Second expected part** image
> 
> I observed the following incorrect behavior:
> 
> **With PR** image
> 
> image
> 
> Could you please check?
> 
> Many thanks,

i tried on 1.7.7.3 and works good

**Pull request closed without merging**